File: orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/domain/JobId.java
package com.sequenceiq.cloudbreak.orchestrator.salt.domain;

public class JobId {

    private final String jobId;

    private JobId(String jobId) {
        this.jobId = jobId;
    }

    public static JobId jobId(String id) {
        return new JobId(id);
    }

    public String getJobId() {
        return jobId;
    }

    @Override
    public String toString() {
        return "JobId{"
                + "jobId='" + jobId + '\''
                + '}';
    }
}

File: orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/domain/JobId.kt
package com.sequenceiq.cloudbreak.orchestrator.salt.domain

class JobId private constructor(val jobId: String) {

    override fun toString(): String {
        return "JobId{"
        +"jobId='" + jobId + '\''
        +'}'
    }

    companion object {

        fun jobId(id: String): JobId {
            return JobId(id)
        }
    }
}
